controller.go

```diff
@@ -22,9 +22,57 @@ import (
 	"net/http"
 	"strconv"
 
+	caoapi "cloud-android-orchestration/api/v1"
+
 	"github.com/gorilla/mux"
 )
 
+type ErrorMsg struct {
+	Error string `json:"error"`
+}
+
+type InfraConfig struct {
+	IceServers []IceServer `json:"ice_servers"`
+}
+type IceServer struct {
+	URLs []string `json:"urls"`
+}
+
+type HTTPHandler func(http.ResponseWriter, *http.Request) error
+
+type DeviceFilesRequest struct {
+	devId string
+	path  string
+	w     http.ResponseWriter
+	r     *http.Request
+}
+
+type SignalingServer interface {
+	// These endpoints in the SignalingServer return the (possibly modified)
+	// response from the Host Orchestrator and the status code if it was
+	// able to communicate with it, otherwise it returns an error.
+	NewConnection(msg caoapi.NewConnMsg, user UserInfo) (*caoapi.SServerResponse, error)
+	Forward(id string, msg caoapi.ForwardMsg, user UserInfo) (*caoapi.SServerResponse, error)
+	Messages(id string, start int, count int, user UserInfo) (*caoapi.SServerResponse, error)
+
+	// Forwards the reques to the device's server unless it's a for a file that
+	// the signaling server needs to serve itself.
+	ServeDeviceFiles(params DeviceFilesRequest, user UserInfo) error
+}
+
+type DeviceDesc struct {
+	// The (internal) network address of the host where the cuttlefish device is
+	// running. The address can either be an IPv4, IPv6 or a domain name.
+	Addr string
+	// The id under which the cuttlefish device is registered with the host
+	// orchestrator (can be different from the id used in the cloud orchestrator)
+	LocalId string
+}
+
+type InstanceManager interface {
+	DeviceFromId(name string, user UserInfo) (DeviceDesc, error)
+}
```

Comment:
These interfaces don't belong in controller.go, they should be in a different file also under api/v1 (see the other comment about moving controller.go).

Reply:
moved to back to types.go.